### PR TITLE
Added a new package: package-require

### PR DESCRIPTION
**Pull request by SpringHan**
### Brief summary of what the package does

Package-Require is a emacs package manager.It's light and has some nice features.For example, it can clone the remote git repositories and help user to take control of them.

### Direct link to the package repository

https://github.com/SpringHan/package-require

### Your association with the package

I'm the maintrainer.

### Relevant communications with the upstream package maintainer

**None Needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [ ] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

This package only requires emacs itself.  Keep updating.  Thanks.

### CODE REVIEW

**Comment by alphapapa** [comment]
Considering the abundance of existing tools like this, and considering that `use-package` is soon to be merged into Emacs, it's questionable whether a package like this is very useful.  At least, you should compare it with tools like `use-package` and `leaf` and explain why yours is a better choice than theirs.  If you really don't like `use-package` for some reason, @conao3 has put a lot of work into making `leaf` a reasonable alternative to it, and he and John are discussing ways to further improve `use-package` in some similar ways.

If you insist on publishing your own, slightly different, load-and-configure-this-package macro, I strongly recommend giving it a distinctive name, like Naoya did for `leaf`.  The name `package-require` sounds generic and so nearly implies that it's part of Emacs.  I could easily imagine a new user becoming quite confused by trying to distinguish `require` and `use-package` and `package-require`.

**Comment by terlar** [comment]
To add to the list, there is also [req-package](https://melpa.org/#/req-package) (wrapper around `use-package`)

**Comment by SpringHan** [comment]
This package is lighter than `use-package` and `leaf`, and it can use packages at remote git repository.  
Now, these are its advantages.  
Thanks.

**Comment by SpringHan** [comment]
And It was renamed `gpack`.

**Comment by alphapapa** [comment]
> and it can use packages at remote git repository.

So it includes functionality to clone git repositories and install packages from them?  That doesn't sound "lighter" than `use-package`.  It sounds like a violation of separation of concern.  See, e.g. `quelpa-use-package`, which integrates with `use-package` forms and provides functionality to install packages from source code repos, while doing so in a separate package.

There are already *so many* packages providing these kinds of tools, each one slightly different and slightly overlapping with another.  Even `leaf`, one of the more powerful ones, may be merged with `use-package`, if John and Naoya can work it out.  It doesn't seem like it would be in the best interests of MELPA users for yet another slightly different tool to muddy the waters at this point.  With 5,000 packages now, discovery and choice overload are two of the most serious challenges for new Emacs and MELPA users to overcome.  Every package like this, which is competing with tried-and-true "giants" of the field, will eventually confuse some new user who will wonder, "Which one should I use?" or who will use an inferior one and think it's state-of-the-art.

If you want to roll your own tool, that's fine, and you should share it and publish it freely.  But it doesn't necessarily mean it needs to be on MELPA for unsuspecting users to have to discern between.

Do you know what I mean?

**Comment by SpringHan** [comment]
OK, I know.Thanks for your advice.

**Comment by alphapapa** [comment]
To be clear, I'm not MELPA, so it's not my decision whether to accept your package.  Steve and/or Chris may disagree with me.  So you shouldn't cancel your PR just because of what I say--only if you truly agree with me.

Anyway, if you're not satisfied with, e.g. `use-package`, `quelpa`, etc, I'd encourage you to consider contributing to them, and that way everyone benefits from your work.

**Comment by SpringHan** [comment]
I think you're right. Now `gpack` has no other adventage relative to `use-package` or `leaf`. And I'm new in Emacs, there're other things I need to learn.  
This package is comes from a simple function writing by me, inspirited by `use-package` when I first use Emacs. So as I wrote this package, I saw it as a package for myself.  
So there maybe some things you don't think very good. And because I'm a student, I only can maintain it on weekends.

In summary, I think I'll not put it into melpa. I'll write real useful package and then try to do it.  
Thanks for all of you.

**Comment by alphapapa** [comment]
Okay, that sounds good.  Thanks for offering your contribution, and I look forward to seeing your future work.
